### DISCUSSION TITLE
Embed GoPro HERO9 with an Android app

### DISCUSSION THREAD

**Discussion by dewtom** [gopro/OpenGoPro#33] [Q&A] [answered]
Hello,
Our team are implementing an Android app, which wants to use GoPro HERO9 as an external device to auto capturing images.
So we need to get some control of it. But there isn't an official API for Kotlin. What methods do you recommend to us? Are you going to implement that in your future release? Or we can just wrap it to Kotlin by ourselves.

Thanks!
Iris

**Comment by tcamise-gpsw** [comment, marked as answer]
Hello. I'm not familiar with Kotlin. But are you asking for an Open GoPro library of some sort to use with Kotlin on Android? If so, no there are no plans to make this. 

Just to be clear, Open GoPro is really just documentation of the camera's BLE and HTTP interfaces. There are some demos / tutorials that show how to do this in various languages in the repo but these are just meant as starting points for the user; not officially supported for production, etc. 

In any case, it would be nice to have an Android demo in here. If you make one, feel free to open a PR to merge it in 👍

**Comment by deleteddreams** [comment]
> Hello, Our team are implementing an Android app, which wants to use GoPro HERO9 as an external device to auto capturing images. So we need to get some control of it. But there isn't an official API for Kotlin. What methods do you recommend to us? Are you going to implement that in your future release? Or we can just wrap it to Kotlin by ourselves.
> 
> Thanks! Iris

Hi,
Did you find a solution? I am having some issue

**Comment by tcamise-gpsw** [comment]
Hello. The [tutorials](https://gopro.github.io/OpenGoPro/tutorials/) now support Kotlin as of last month. The tutorials include a simple app.
Take a look at those and let me know if you have any issues.
